pipeline/backend/types/engine.go

```diff
@@ -47,3 +47,9 @@ type Engine interface {
 	// DestroyWorkflow the workflow environment.
 	DestroyWorkflow(ctx context.Context, conf *Config, taskUUID string) error
 }
+
+// EngineInfo represents the reported information of a loaded engine
+type EngineInfo struct {
+	Platform string
+	Backend  string
```

Comment:
Why not stick and directly use `backend.Name()` seems kinda duplicated.

Reply:
I would just remove it from `EngineInfo` and use `backend.Name()` in the places as before.